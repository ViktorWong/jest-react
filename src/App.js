import React from "react";

function App() {
  return <div className="App" title="hello" data-name="App">hello world</div>;
}

export default App;
