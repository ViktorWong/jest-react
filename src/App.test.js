import React from 'react';
import App from './App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


// 单元测试适合用shallow,集成测试适合用mount
test('renders', () => {
   const wrapper = Enzyme.shallow(<App/>)
   expect(wrapper.find('.App').length).toBe(1)
   console.log(wrapper.debug())
   expect(wrapper.find('.App').prop('title')).toBe('hello')
});

test('renders with data-name', () => {
  const wrapper = Enzyme.shallow(<App/>)
  console.log(wrapper.find('[data-name="App"]'))
  expect(wrapper.find('[data-name="App"]')).toExist()
  expect(wrapper.find('[data-name="App"]')).toHaveProp('title','hello')
  expect(wrapper).toMatchSnapshot();
});
