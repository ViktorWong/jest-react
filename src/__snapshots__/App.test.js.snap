// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data-name 1`] = `
<div
  className="App"
  data-name="App"
  title="hello"
>
  hello world
</div>
`;
